lib/methods/findAll.js

```diff
@@ -1,8 +1,17 @@
+var _ = require('lodash');
 /**
  * Finds all model instances.  A maximum of 1000 models are returned.
  * @param {Arrow.Model} Model The model class being updated.
  * @param {Function} callback Callback passed an Error object (or null if successful) and the models.
  */
 exports.findAll = function (Model, callback) {
-	Model._query({limit: 1000}, callback, Model);
+	var options = {
+		limit: 1000
+	};
+
+	if(Model.metadata && _.isNumber(Model.metadata.responseJsonDepth)) {
+		options.response_json_depth = Model.metadata.responseJsonDepth | 0;
+	}
```

Comment:
Not sure i understand why this block still necessary when it's already in the `invoke` method? `Model._query` just uses the `invoke` method.